### PR TITLE
python310Packages.bentoml: init at 1.0.22

### PR DESCRIPTION
**Pull request by happysalada**
###### Description of changes

proposal for bentoml.
@natsukium in case you are interested.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 245377` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python310Packages.bentoml</li>
    <li>python310Packages.bentoml.dist</li>
    <li>python311Packages.bentoml</li>
    <li>python311Packages.bentoml.dist</li>
  </ul>
</details>

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 245377` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python310Packages.bentoml</li>
    <li>python310Packages.bentoml.dist</li>
    <li>python311Packages.bentoml</li>
    <li>python311Packages.bentoml.dist</li>
  </ul>
</details>

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 245377` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.bentoml</li>
    <li>python310Packages.bentoml.dist</li>
    <li>python311Packages.bentoml</li>
    <li>python311Packages.bentoml.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/bentoml/default.nix [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  ];
+
+  passthru.optional-dependencies = optional-dependencies;
+
+  pythonImportsCheck = [ "bentoml" ];
+
+  # tests try to install additional requirements with pip
+  # I'm sure there is a way to override that at some point
+  doCheck = false;
+
```
Comment (natsukium) [inline pkgs/development/python-modules/bentoml/default.nix:172]:
```suggestion
```

File: pkgs/development/python-modules/bentoml/default.nix [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+  nativeCheckInputs = [
+    pytestCheckHook
+    pandas
+    pydantic
+    scikit-learn
+    yamllint
+    coverage
```
Comment (natsukium) [inline pkgs/development/python-modules/bentoml/default.nix:179]:
```suggestion
```

File: pkgs/development/python-modules/bentoml/default.nix [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    lxml
+    orjson
+    pytest-cov
```
Comment (natsukium) [inline pkgs/development/python-modules/bentoml/default.nix:182]:
```suggestion
```

File: pkgs/development/python-modules/bentoml/default.nix [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    pytest-asyncio
+    pillow
+    fastapi
+    starlette
+  ];
```
Comment (natsukium) [inline pkgs/development/python-modules/bentoml/default.nix:187]:
```suggestion
  ] ++ optional-dependencies.grpc;
```

File: pkgs/development/python-modules/bentoml/default.nix [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+
```
Comment (natsukium) [inline pkgs/development/python-modules/bentoml/default.nix:188]:
```suggestion

  preCheck = ''
    # required for CI testing
    # https://github.com/bentoml/BentoML/pull/4056/commits/66302b502a3f4df4e8e6643d2afefefca974073e
    export GITHUB_ACTIONS=1
  '';
  
  disabledTestPaths = [
    "tests/e2e"
    "tests/integration"
  ];
  
  disabledTests = [
    # flaky test
    "test_store"
  ];
  
```

File: pkgs/development/python-modules/bentoml/default.nix [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+  meta = with lib; {
+    description = "Build Production-Grade AI Applications";
+    homepage = "https://github.com/bentoml/BentoML";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ happysalada ];
```
Comment (natsukium) [inline pkgs/development/python-modules/bentoml/default.nix:193]:
```suggestion
    maintainers = with maintainers; [ happysalada natsukium ];
```
:)

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 245377` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.bentoml</li>
    <li>python310Packages.bentoml.dist</li>
    <li>python311Packages.bentoml</li>
    <li>python311Packages.bentoml.dist</li>
  </ul>
</details>

**Comment by happysalada** [comment]
I think ive taken care of all your comments

Are you interested in openllm as well ? I would be happy to either let you package it and be a co maintainer or make a PR myself and tag you. Both work for me.

**Comment by natsukium** [comment]
If you are not working on it yet, I have a draft for openllm and will submit a PR.

**Review by natsukium — APPROVED** [review]

**Pull request merged**